main.tf

```diff
@@ -0,0 +1,234 @@
+      DENIED_REGIONS              = length(var.config.denied-regions) > 0 ? join(", ", var.config.denied-regions) : null
+      DISABLE_BANNER              = true,
+      DISABLE_SPINNER             = true
+      EXPORT_METRICS_ONLY         = var.config.export-metrics-only
+      EXPORT_PATH                 = var.config.export-path
+      FRAMEWORKS                  = length(var.config.frameworks) > 0 ? join(", ", var.config.frameworks) : null
+      LOG_LEVEL                   = var.config.log-level
+      ORGANIZATIONS_ZONE_NAME     = var.config.organizations-zone-name
+      REGION                      = var.config.region != null ? var.config.region : data.aws_region.current.name
+      REPORT_CLOSED_FINDINGS_DAYS = var.config.report-closed-findings-days
+      REPORT_SUPPRESSED_FINDINGS  = var.config.report-suppressed-findings
+      SECURITY_HUB_QUERY_FILTER   = var.config.security-hub-query-filter
+      SINGLE_ACCOUNT_ID           = var.config.single-account-id
+      TO_JSON                     = var.config.to-json
+      VALIDATE_METADATA_FILE      = var.config.validate-metadata-file
+      ZONE_THRESHOLDS             = var.config.zone-thresholds
+      } : {
+      name  = "AWS_LABELER_${name}",
+      value = value
+    } if value != null
+  ]
+
+  s3_export_target    = try(strcontains(var.config["export-path"], "s3://"), false) ? { create : true } : {}
+  single_account_id   = can(var.config["single-account-id"]) ? { create : true } : {}
+  multiple_account_id = anytrue([can(var.config["organizations-zone-name"]), can(var.config["audit-zone-name"])]) ? { create : true } : {}
+}
+
+data "aws_subnet" "selected" {
+  count = var.subnet_ids != null ? 1 : 0
+
+  id = var.subnet_ids[0]
+}
+
+resource "aws_security_group" "default" {
+  #checkov:skip=CKV2_AWS_5: False positive finding, the security group is attached.
+  count = var.subnet_ids != null ? 1 : 0
+
+  name        = local.name
+  description = "Security group for ESC ${local.name}"
+  vpc_id      = data.aws_subnet.selected[0].vpc_id
+  tags        = var.tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_vpc_security_group_egress_rule" "default" {
+  for_each = var.subnet_ids != null && length(var.security_group_egress_rules) != 0 ? { for v in var.security_group_egress_rules : v.description => v } : {}
+
+  cidr_ipv4                    = each.value.cidr_ipv4
+  cidr_ipv6                    = each.value.cidr_ipv6
+  description                  = each.value.description
+  from_port                    = each.value.from_port
+  ip_protocol                  = each.value.ip_protocol
+  prefix_list_id               = each.value.prefix_list_id
+  referenced_security_group_id = each.value.referenced_security_group_id
+  security_group_id            = aws_security_group.default[0].id
+  to_port                      = each.value.to_port
+}
+
+resource "aws_ecs_cluster" "default" {
+  count = var.cluster_arn == null ? 1 : 0
+
+  name = "ecs-scheduled-task"
```

Comment:
should there not be some "energylabeler" or somethign in the name?

Reply:
I don't think the task and the cluster should be or are related

Reply:
That makes sense but if one opts for letting this moduel create the cluster its not going to be re-used by other thigns right? That would not make sense to me tbh

Reply:
default name is now used